cmd/jujud/updateseries/updateseries.go

```diff
@@ -118,38 +139,64 @@ func (c *UpdateSeriesCommand) Run(ctx *cmd.Context) error {
 		return err
 	}
 
+	jujuVersion, err := GetJujuVersion(c.machineAgent, c.dataDir)
+	if err != nil {
+		return err
+	}
+
 	switch {
 	case toInitSys == service.InitSystemUpstart:
 		return errors.NewNotSupported(nil, "downgrade to series using upstart not supported")
 	case toInitSys == fromInitSys && toInitSys == service.InitSystemSystemd:
-		if err = c.copyTools(); err != nil {
+		if err = service.CopyAgentBinary(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries, c.fromSeries, jujuVersion); err != nil {
 			return err
 		} else {
 			ctx.Infof("successfully copied tools and relinked agent tools")
 		}
 		if !c.startAgents {
 			break
 		}
-		if err = c.startAllAgents(ctx); err != nil {
+		if errMachineName, errUnitNames, err = service.StartAllAgents(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries); err != nil {
 			return err
+		} else {
+			for _, unitName := range errUnitNames {
+				ctx.Infof("started %s service", unitName)
+			}
+			ctx.Infof("started %s service", errMachineName)
 		}
 		ctx.Infof("all agents successfully restarted")
 	case toInitSys == service.InitSystemSystemd:
 		errorHappened := false
-		if err = c.writeSystemdAgents(ctx); err != nil {
+		errAgentNames = errAgentNames[:0]
+		if errSysdServiceNames, errSymServiceNames, errAgentNames, err = service.WriteSystemdAgents(c.machineAgent, c.unitAgents, c.dataDir, systemdDir, systemdMultiUserDir, c.toSeries); err != nil {
```

Comment:
Same here - use := to define these variables and limit them to this if statement. (And the same comment about the names for `errSysd...` and errSym...`.)